paperqa/core.py

```diff
@@ -115,3 +116,75 @@ async def map_fxn_summary(
         ),
         llm_result,
     )
+
+
+async def gather_with_batch(
+    matches: list[Text],
+    question: str,
+    prompt_runner: PromptRunner | None,
+    extra_prompt_data: dict[str, str] | None = None,
+    parser: Callable[[str], dict[str, Any]] | None = None,
+    callbacks: list[Callable[[str], None]] | None = None,
+) -> list[tuple[Context, LLMResult]]:
+    """
+    Gathers evidence considering a batch of texts. The completions are obtained using a batch API.
+
+    Args:
+        matches (list[Text]): A list of text matches to gather evidence from.
+        question (str): The question to be answered.
+        prompt_runner (PromptRunner | None): The prompt runner to use for obtaining completions.
```

Comment:
When doing docstrings, if you put type hints in the signature (which you did, and which is the right move), don't duplicate the types in the docstring.

What happens is later on, people will update the type hint in the signature, but miss they were needed to also update the docstring

So make this:

> prompt_runner: Optional prompt runner to use for obtaining completions.

Reply:
Makes sense. Updated it